lib/TasksResource.ts

```diff
@@ -180,49 +223,50 @@ export interface TaskCaptureData {
     headers?: { [header: string]: string };
 }
 
-interface TaskOptimize {
-    operation: 'optimize';
-    data: TaskOptimizeData;
+interface TaskThumbnail {
+    operation: 'thumbnail';
+    data: TaskThumbnailData;
 }
 
-export interface TaskOptimizeData {
+export interface TaskThumbnailData {
     input: string | string[];
-    input_format?: 'jpg' | 'png' | 'pdf';
+    input_format?: string;
+    output_format: 'png' | 'jpg';
     engine?: string;
     engine_version?: string;
     filename?: string;
-    quality?: number;
-    profile?: 'web' | 'print' | 'archive' | 'mrc' | 'max';
+    timeout?: number;
 
     [option: string]: any;
 }
 
-interface TaskThumbnail {
-    operation: 'thumbnail';
-    data: TaskThumbnailData;
-}
-
 interface TaskMetadata {
     operation: 'metadata';
     data: TaskMetadataData;
 }
 
-export interface TaskThumbnailData {
+export interface TaskMetadataData {
     input: string | string[];
     input_format?: string;
-    output_format: 'png' | 'jpg';
     engine?: string;
     engine_version?: string;
-    filename?: string;
+    timeout?: number;
 
     [option: string]: any;
 }
 
-export interface TaskMetadataData {
+interface TaskMetadataWrite {
+    operation: 'metadata/write';
```

Comment:
Is this the correct operation type? I'm not 100 % from the docs.

Reply:
Yes it is correct

Reply:
Perfect!